contracts/ethregistrar/ETHRegistrarController.sol

```diff
@@ -3,176 +3,312 @@ pragma solidity >=0.8.4;
-        keccak256("register(string,address,uint256,bytes32)") ^
-        keccak256("renew(string,uint256)")
+    using BytesUtils for bytes;
+
+    uint256 public constant MIN_REGISTRATION_DURATION = 28 days;
+
+    bytes4 private constant INTERFACE_META_ID =
+        bytes4(keccak256("supportsInterface(bytes4)"));
+    bytes4 private constant COMMITMENT_CONTROLLER_ID =
+        type(commitmentController).interfaceId;
+    bytes4 private constant COMMITMENT_WITH_CONFIG_CONTROLLER_ID =
+        bytes4(
+            keccak256(
+                "register(string,address,uint256,bytes32,address,address)"
+            ) ^
+                keccak256(
+                    "makeCommitment(string,address,bytes32,address,address,bool,int96)"
+                )
+        );
+
+    BaseRegistrarImplementation immutable base;
+    PriceOracle public prices;
+    uint256 public immutable minCommitmentAge;
+    uint256 public immutable maxCommitmentAge;
+    ReverseRegistrar public immutable reverseRegistrar;
+    INameWrapper public immutable nameWrapper;
+
+    mapping(bytes32 => uint256) public commitments;
+
+    event NameRegistered(
+        string name,
+        bytes32 indexed label,
+        address indexed owner,
+        uint256 baseCost,
+        uint256 premium,
+        uint256 expires
     );
-
-    bytes4 constant private COMMITMENT_WITH_CONFIG_CONTROLLER_ID = bytes4(
-        keccak256("registerWithConfig(string,address,uint256,bytes32,address,address)") ^
-        keccak256("makeCommitmentWithConfig(string,address,bytes32,address,address)")
+    event NameRenewed(
+        string name,
+        bytes32 indexed label,
+        uint256 cost,
+        uint256 expires
     );
-
-    BaseRegistrarImplementation base;
-    PriceOracle prices;
-    uint public minCommitmentAge;
-    uint public maxCommitmentAge;
-
-    mapping(bytes32=>uint) public commitments;
-
-    event NameRegistered(string name, bytes32 indexed label, address indexed owner, uint cost, uint expires);
-    event NameRenewed(string name, bytes32 indexed label, uint cost, uint expires);
     event NewPriceOracle(address indexed oracle);
 
-    constructor(BaseRegistrarImplementation _base, PriceOracle _prices, uint _minCommitmentAge, uint _maxCommitmentAge) public {
+    constructor(
+        BaseRegistrarImplementation _base,
+        PriceOracle _prices,
+        uint256 _minCommitmentAge,
+        uint256 _maxCommitmentAge,
+        ReverseRegistrar _reverseRegistrar,
+        INameWrapper _nameWrapper
+    ) {
         require(_maxCommitmentAge > _minCommitmentAge);
 
         base = _base;
         prices = _prices;
         minCommitmentAge = _minCommitmentAge;
         maxCommitmentAge = _maxCommitmentAge;
+        reverseRegistrar = _reverseRegistrar;
+        nameWrapper = _nameWrapper;
     }
 
-    function rentPrice(string memory name, uint duration) view public returns(uint) {
+    function rentPrice(string memory name, uint256 duration)
```

Comment:
Should we add a pass-through for the inverse too, now that we have it?

Reply:
Or maybe make this internal, and people can just fetch the PriceOracle through the public variable.